context/app/static/js/components/workspaces/ConfirmDeleteWorkspacesDialog/ConfirmDeleteWorkspacesDialog.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback } from 'react';
+
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Divider from '@mui/material/Divider';
+import IconButton from '@mui/material/IconButton';
+
+import CloseRounded from '@mui/icons-material/CloseRounded';
+
+import { useSnackbarActions } from 'js/shared-styles/snackbars';
+import { SelectedItems } from 'js/hooks/useSelectItems';
+import { generateCommaList } from 'js/helpers/functions';
+
+import { MergedWorkspace } from '../types';
+
+interface ConfirmDeleteWorkspacesDialogProps {
+  dialogIsOpen: boolean;
+  handleClose: () => void;
+  handleDeleteWorkspace: (workspaceId: number) => Promise<void>;
+  selectedWorkspaceIds: SelectedItems;
+  workspacesList: MergedWorkspace[];
+}
+export default function ConfirmDeleteWorkspacesDialog({
+  dialogIsOpen,
+  handleClose,
+  handleDeleteWorkspace,
+  selectedWorkspaceIds,
+  workspacesList,
+}: ConfirmDeleteWorkspacesDialogProps) {
+  const { toastError, toastSuccess } = useSnackbarActions();
+
+  const selectedWorkspaceNames = Array.from(selectedWorkspaceIds).map((id) => {
+    const workspace = workspacesList.find((w) => w.id === Number(id));
+    return workspace ? workspace.name : '';
+  });
+
+  const handleDeleteAndClose = useCallback(() => {
+    const workspaceIds = [...selectedWorkspaceIds];
+
+    Promise.all(workspaceIds.map((workspaceId) => handleDeleteWorkspace(Number(workspaceId))))
+      .then(() => {
+        toastSuccess(`Successfully deleted workspaces: ${generateCommaList(selectedWorkspaceNames)}`);
+        selectedWorkspaceIds.clear();
+      })
+      .catch((e) => {
+        toastError(`Error deleting workspaces: ${generateCommaList(selectedWorkspaceNames)}`);
```

Comment:
Can we just call `generateCommaList` once and use its output here and in the jsx below?